lab-13-JR/route/recipe-route.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+const Router = require('express').Router;
+const jsonParser = require('body-parser').json();
+const Recipe = require('../model/recipe');
+
+const recipeRouter = module.exports = new Router();
+
+recipeRouter.post('/api/recipe', jsonParser, function(req, res, next) {
+  req.body.timestamp = new Date();
+  console.log(req.body);
+  new Recipe(req.body).save()//returns a promise
+  .then(recipe => res.json(recipe))
+  .catch(next);
+});
+
+recipeRouter.get('/api/recipe/:id', function(req, res, next) {
+  Recipe.findById(req.params.id)
+  .then(recipe => res.json(recipe))
+  .catch(next);
+});
+
+recipeRouter.get('/api/recipe', function(req, res, next) {
+  Recipe.find().toArray() //this is happening synchronously and toArray is trying to work with something that isn't done yet. Promisify, or use toArray() as a cb function?
```

Comment:
What happens when you remove the `.toArray()` and just send back the collection of recipes?